sessions/debugging.qmd

```diff
@@ -1,24 +1,192 @@
 # Checking correctness of code: Debugging
 
--   Using basic debugging strategies ('breakpoints', simple `print()` or
-    `stop()` calls, `debug()`, `browser()`)
--   Can take some from the intermediate course (I think that's where it
-    is)
-
 {{< include ../includes/_wip.qmd >}}
 
 ```{r setup}
 #| include: false
 ```
 
-<!-- TODO: Add an introduction here -->
+**Introduction**
+
+Debugging is one of activities that seems really scary and difficult,
+but once you try it and use it, is not nearly as intimidating as it
+seemed. In this session we will focus on using RStudio's interactive
+debugger to debug code.
 
 **Session objectives**:
 
--   Enter the interactive debugger and use it to explore why unexpected
-    behavior is occurring
+-   Understand what debugging is.
```

Comment:
```suggestion
-   Describe what formal debugging is in the context of package development in RStudio.
```

In general, you want to avoid using "understand" and "learn" in learning objectives, because they don't help the learner know what to expect in a more tangible sense.